### PR TITLE
Cumulative Update for Keras 2.2.0 and some fixes and Explanations in the README.md

### PR DESCRIPTION
**Pull request by Pyrestone**
This Pull request includes fixes for the issues addressed in #102 #101, #96, #94 and #76 
along with the bug in farizrahman4u/seq2seq#177 and farizrahman4u/seq2seq#250 
(both caused by calls to the deprecated  `keras.engine.topology` class.)

### CODE REVIEW

**Comment by Pyrestone** [comment]
While waiting for a response from @farizrahman4u, you can use the installation instructions from [pyrestone/recurrentshop](https://www.github.com/pyrestone/recurrentshop) to install the library for keras 2.2.0

**Comment by Pyrestone** [comment]
I added another fix for the Tensorflow teacher-forcing bug, fixing #81 and making the tf-teacher-force branch obsolete.

**Comment by farizrahman4u** [comment]
Thanks for the effort. Couple of small changes required.

File: README.md [hunk 1]
```diff
@@ -34,17 +34,35 @@ h_t = add([Dense(10)(x_t), Dense(10, use_bias=False)(h_tm1)])
...
+
+# Build a Keras Model using our RNN layer
+# input dimensions are (Time_steps, Depth)
+x = Input(shape=(7,5))
+y = rnn(x)
+model = Model(x, y)
 
 # Run the RNN over a random sequence
+# Don't forget the batch shape when calling the model!
+out=model.predict(np.random.random((1, 7, 5)))
```
Comment (farizrahman4u) [inline README.md]: Spaces around =

File: README.md [hunk 2]
```diff
@@ -34,17 +34,35 @@ h_t = add([Dense(10)(x_t), Dense(10, use_bias=False)(h_tm1)])
...
+# to get one output per input sequence element, set return_sequences=True
+rnn2 = RecurrentModel(input=x_t, initial_states=[h_tm1], output=h_t, final_states=[h_t],return_sequences=True)
+
+# Time_steps can also be None to allow variable Sequence Length
+# Note that this is not compatible with unroll=True
+x = Input(shape=(None ,5))
+y = rnn2(x)
+model2 = Model(x, y)
+
+out2=model2.predict(np.random.random((1, 7, 5)))
```
Comment (farizrahman4u) [inline README.md]: Spaces around =

**Comment by farizrahman4u** [comment]
Lgtm, Thanks!

**Comment by ty5491003** [comment]
Thank you for your work!

**Pull request merged**